src/map/latent_effect_container.cpp

```diff
@@ -1040,52 +1047,59 @@ bool CLatentEffectContainer::ProcessLatentEffect(CLatentEffect& latentEffect)
         case LATENT::NATION_CONTROL:
         {
             // player is logging in/zoning
-            if (m_POwner->loc.zone == nullptr)
+            // checking the state of the user's current zone and
+            // the destination zone in tandem seems to work.
+            if (m_POwner->loc.zone == nullptr && static_cast<uint16_t>(m_POwner->loc.destination) == 0)
             {
                 break;
             }
 
-            auto region      = m_POwner->loc.zone->GetRegionID();
+            // Grab the user's destination if they're zoning.
+            // Otherwise, grab their current zone.
+            auto region      = m_POwner->loc.zone == nullptr ? zoneutils::GetCurrentRegion(m_POwner->loc.destination) : zoneutils::GetCurrentRegion(playerZoneID);
             auto hasSignet   = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SIGNET);
             auto hasSanction = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SANCTION);
             auto hasSigil    = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SIGIL);
-
+            auto regionAlwaysOutOfControl = zoneutils::IsAlwaysOutOfNationControl(region);
             switch (latentEffect.GetConditionsValue())
             {
                 case 0:
                     // under own nation's control
-                    expression = region < REGION_TYPE::WEST_AHT_URHGAN && conquest::GetRegionOwner(region) == m_POwner->profile.nation &&
+                    expression = region < REGION_TYPE::WEST_AHT_URHGAN && (!regionAlwaysOutOfControl || conquest::GetRegionOwner(region) == m_POwner->profile.nation) &&
                                  (hasSignet || hasSanction || hasSigil);
                     break;
                 case 1:
                     // outside of own nation's control
-                    expression = region < REGION_TYPE::WEST_AHT_URHGAN && m_POwner->profile.nation != conquest::GetRegionOwner(region) &&
+                    expression = region < REGION_TYPE::WEST_AHT_URHGAN && (regionAlwaysOutOfControl || m_POwner->profile.nation != conquest::GetRegionOwner(region)) &&
                                  (hasSignet || hasSanction || hasSigil);
                     break;
             }
             break;
         }
         case LATENT::ZONE_HOME_NATION:
         {
+
             // player is logging in/zoning
-            if (m_POwner->loc.zone == nullptr)
+            // checking the state of the user's current zone and
+            // the destination zone in tandem seems to work.
+            if (m_POwner->loc.zone == nullptr && static_cast<uint16_t>(m_POwner->loc.destination) == 0)
```

Comment:
If we're still inside `CLatentEffectContainer::ProcessLatentEffect(CLatentEffect& latentEffect)`didnt you already check zone validity at the top? `(m_POwner->loc.zone == nullptr &`

Reply:
```
uint16 CBaseEntity::getZone() const
{
    return loc.zone != nullptr ? (uint16)loc.zone->GetID() : (uint16)loc.destination;
}
```

Reply:
I removed all redundant checks